### DISCUSSION TITLE
如何让adaptive-sharpen只在缩小(downscaling)时生效

### DISCUSSION THREAD

**Discussion by yeezylife** [hooke007/mpv_PlayKit#142] [Q&A 问答] [answered]
[adaptive-sharpen的评论里](https://gist.github.com/igv/8a77e4eb8276753b54bb94c1c50c317e)有个人做了很多测试,得出的结果是luma版本的adaptive-sharpen 0.1强度+lanczos 在缩小的时候表现最好

不过adaptive-sharpen在upscaling的时候也会生效,这就很麻烦,那样不同的尺寸的视频我就需要不断的开启和关闭adaptive-sharpen.(我只想在缩小的时候启用adaptive-sharpen)

于是我自己根据其他shaders里的内容改了改,在adaptive-sharpen里增加/修改了这些:

`//!HOOK LUMA`
`//!WHEN OUTPUT.w LUMA.w / 1.000 < OUTPUT.h LUMA.h / 1.000 < *`
`//!BIND HOOKED`

这样是不是对的呀,我的期望是adaptive-sharpen只在缩小的时候生效...我自己都不知道自己在干嘛...但测试了一下好像能用

**Comment by hooke007** [comment, marked as answer]
试试这个
```
//!WHEN OUTPUT.w OUTPUT.h * LUMA.w LUMA.h * / 1.0 <
```

**Reply by yeezylife** [reply]
我试了试效果似乎是一样的.我的这个是根据FSRCNNX里的这段改来的...
`//!HOOK LUMA`
`//!WHEN OUTPUT.w LUMA.w / 1.300 > OUTPUT.h LUMA.h / 1.300 > *`

**Reply by hooke007** [reply]
没毛病啊换数字和符号就完事了

**Reply by yeezylife** [reply]
另外就是我猜想出现这种要用adaptivesharpen替代SSIMDownscaler的情况,从侧面说明了SSIMDownscaler效果不够理想,就我自己的测试,清晰度没明显增加,artifacts倒是明显增加了...

**Reply by yeezylife** [reply]
我这样改还是有个问题...FSRCNNX放大两倍之后再缩小的时候,adaptivesharpen不会生效...

**Reply by yeezylife** [reply]
改成你的写法,FSRCNNX放大两倍之后再缩小,adaptivesharpen也能生效了,大佬就是不一样!!
